### ISSUE TITLE
Frequent DISCONNECT events

### ISSUE THREAD

**Issue by sdwr98** [nzapponi/homebridge-simplisafe3#55]
First off - thanks so much for this plugin! It's really improved my HomeKit setup. 

I'm running 1.4.0 on Homebridge on a Raspberry Pi.  When I start everything up, things go well - but at some point, usually a day or two later, I start getting tons of DISCONNECTS: 

```
Jan 23 17:09:10 raspberrypi homebridge[240]: [1/23/2020, 17:09:10] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 23 17:09:10 raspberrypi homebridge[240]: [1/23/2020, 17:09:10] [Home Alarm] Real time events disconnected.
Jan 23 17:09:10 raspberrypi homebridge[240]: [1/23/2020, 17:09:10] [Home Alarm] Listening to alarm events...
Jan 23 17:09:10 raspberrypi homebridge[240]: [1/23/2020, 17:09:10] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 23 17:09:10 raspberrypi homebridge[240]: [1/23/2020, 17:09:10] [Home Alarm] Real time events disconnected.
Jan 23 17:09:10 raspberrypi homebridge[240]: [1/23/2020, 17:09:10] [Home Alarm] Listening to alarm events...
Jan 23 17:09:11 raspberrypi homebridge[240]: [1/23/2020, 17:09:11] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 23 17:09:11 raspberrypi homebridge[240]: [1/23/2020, 17:09:11] [Home Alarm] Real time events disconnected.
```

And it spins in a tight loop like that for a while.  Then it'll settle down for a while, but then get a lot of disconnect events again.  

When the system is in that state, I usually cannot control or query the Simplisafe sensors via HomeKit. 

Is this indicative of a bug, or external factors?  Happy to help debug this...

**Comment by nikonratm** [comment]
Hi @sdwr98, it’s unlikely this is actually to do with the plugin itself as “disconnect” events are just the socket connection (which is a kind of like an “always-on” web connection) events. There are multiple possibilities for why this might happen but usually it’s a connectivity issue related to the (remote) server. Is your internet connection at all flaky? What device is Homebridge running on? Again, haven’t heard this from others which also suggests it’s something specific to your setup...

**Comment by sdwr98** [comment]
So I put a little extra debugging in :  here's what the root causes of the disconnects are.  Does this lead to anything else interesting?  My home internet is pretty rock solid, as far as I can tell. 

I'm running on a Raspberry Pi Zero W.  

```
Jan 29 15:34:15 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:15 raspberrypi homebridge[5843]: [1/29/2020, 15:34:15] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:15 raspberrypi homebridge[5843]: [1/29/2020, 15:34:15] [Home Alarm] Real time events disconnected.
Jan 29 15:34:15 raspberrypi homebridge[5843]: [1/29/2020, 15:34:15] [Home Alarm] Listening to alarm events...
Jan 29 15:34:15 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:15 raspberrypi homebridge[5843]: [1/29/2020, 15:34:15] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:15 raspberrypi homebridge[5843]: [1/29/2020, 15:34:15] [Home Alarm] Real time events disconnected.
Jan 29 15:34:15 raspberrypi homebridge[5843]: [1/29/2020, 15:34:15] [Home Alarm] Listening to alarm events...
Jan 29 15:34:16 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:16 raspberrypi homebridge[5843]: [1/29/2020, 15:34:16] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:16 raspberrypi homebridge[5843]: [1/29/2020, 15:34:16] [Home Alarm] Real time events disconnected.
Jan 29 15:34:16 raspberrypi homebridge[5843]: [1/29/2020, 15:34:16] [Home Alarm] Listening to alarm events...
Jan 29 15:34:16 raspberrypi homebridge[5843]: caught error in socket: Invalid namespace
Jan 29 15:34:16 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:16 raspberrypi homebridge[5843]: [1/29/2020, 15:34:16] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:16 raspberrypi homebridge[5843]: [1/29/2020, 15:34:16] [Home Alarm] Real time events disconnected.
Jan 29 15:34:16 raspberrypi homebridge[5843]: [1/29/2020, 15:34:16] [Home Alarm] Listening to alarm events...
Jan 29 15:34:16 raspberrypi homebridge[5843]: caught error in socket: Invalid namespace
Jan 29 15:34:17 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:17 raspberrypi homebridge[5843]: [1/29/2020, 15:34:17] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:17 raspberrypi homebridge[5843]: [1/29/2020, 15:34:17] [Home Alarm] Real time events disconnected.
Jan 29 15:34:17 raspberrypi homebridge[5843]: [1/29/2020, 15:34:17] [Home Alarm] Listening to alarm events...
Jan 29 15:34:18 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:18 raspberrypi homebridge[5843]: [1/29/2020, 15:34:18] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:18 raspberrypi homebridge[5843]: [1/29/2020, 15:34:18] [Home Alarm] Real time events disconnected.
Jan 29 15:34:18 raspberrypi homebridge[5843]: [1/29/2020, 15:34:18] [Home Alarm] Listening to alarm events...
Jan 29 15:34:18 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:18 raspberrypi homebridge[5843]: [1/29/2020, 15:34:18] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:18 raspberrypi homebridge[5843]: [1/29/2020, 15:34:18] [Home Alarm] Real time events disconnected.
Jan 29 15:34:18 raspberrypi homebridge[5843]: [1/29/2020, 15:34:18] [Home Alarm] Listening to alarm events...
Jan 29 15:34:19 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:19 raspberrypi homebridge[5843]: [1/29/2020, 15:34:19] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:19 raspberrypi homebridge[5843]: [1/29/2020, 15:34:19] [Home Alarm] Real time events disconnected.
Jan 29 15:34:19 raspberrypi homebridge[5843]: [1/29/2020, 15:34:19] [Home Alarm] Listening to alarm events...
Jan 29 15:34:19 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:19 raspberrypi homebridge[5843]: [1/29/2020, 15:34:19] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:19 raspberrypi homebridge[5843]: [1/29/2020, 15:34:19] [Home Alarm] Real time events disconnected.
Jan 29 15:34:19 raspberrypi homebridge[5843]: [1/29/2020, 15:34:19] [Home Alarm] Listening to alarm events...
Jan 29 15:34:19 raspberrypi homebridge[5843]: caught error in socket: Invalid namespace
Jan 29 15:34:20 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:20 raspberrypi homebridge[5843]: [1/29/2020, 15:34:20] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:20 raspberrypi homebridge[5843]: [1/29/2020, 15:34:20] [Home Alarm] Real time events disconnected.
Jan 29 15:34:20 raspberrypi homebridge[5843]: [1/29/2020, 15:34:20] [Home Alarm] Listening to alarm events...
Jan 29 15:34:21 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:21 raspberrypi homebridge[5843]: [1/29/2020, 15:34:21] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:21 raspberrypi homebridge[5843]: [1/29/2020, 15:34:21] [Home Alarm] Real time events disconnected.
Jan 29 15:34:21 raspberrypi homebridge[5843]: [1/29/2020, 15:34:21] [Home Alarm] Listening to alarm events...
Jan 29 15:34:21 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:21 raspberrypi homebridge[5843]: [1/29/2020, 15:34:21] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:21 raspberrypi homebridge[5843]: [1/29/2020, 15:34:21] [Home Alarm] Real time events disconnected.
Jan 29 15:34:21 raspberrypi homebridge[5843]: [1/29/2020, 15:34:21] [Home Alarm] Listening to alarm events...
Jan 29 15:34:22 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:22 raspberrypi homebridge[5843]: [1/29/2020, 15:34:22] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:22 raspberrypi homebridge[5843]: [1/29/2020, 15:34:22] [Home Alarm] Real time events disconnected.
Jan 29 15:34:22 raspberrypi homebridge[5843]: [1/29/2020, 15:34:22] [Home Alarm] Listening to alarm events...
Jan 29 15:34:23 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Real time events disconnected.
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Listening to alarm events...
Jan 29 15:34:23 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Real time events disconnected.
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Listening to alarm events...
Jan 29 15:34:23 raspberrypi homebridge[5843]: caught disconnect in socket: transport close
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Real time events disconnected.
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Listening to alarm events...
Jan 29 15:34:23 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Real time events disconnected.
Jan 29 15:34:23 raspberrypi homebridge[5843]: [1/29/2020, 15:34:23] [Home Alarm] Listening to alarm events...
Jan 29 15:34:24 raspberrypi homebridge[5843]: caught error in socket: Not authorized
Jan 29 15:34:24 raspberrypi homebridge[5843]: [1/29/2020, 15:34:24] [Home Alarm] Received new event from alarm: DISCONNECT
Jan 29 15:34:24 raspberrypi homebridge[5843]: [1/29/2020, 15:34:24] [Home Alarm] Real time events disconnected.
Jan 29 15:34:24 raspberrypi homebridge[5843]: [1/29/2020, 15:34:24] [Home Alarm] Listening to alarm events...
```

**Comment by nikonratm** [comment]
Yes, it suggests you are getting locked out by SS, "rate limited" which is something all of us have been experiencing, annoyingly. The lead plugin author is working on a solution that will hopefully help but really the only solution is to wait until SS remove the lockout on your IP (e.g. shut down Homebridge or, once the plugin is updated, it will essentially do this for you).

Will be closed by #60

**Comment by sdwr98** [comment]
That's what I was suspecting.  Thanks for taking a look!
